discovery/pkg/sync/queue.go

```diff
@@ -110,17 +108,21 @@ func (sq *Queue) processNextWorkItem() bool {
 	action, ok := obj.(Action)
 	if !ok {
 		sq.Workqueue.Forget(obj)
-		sq.Metrics.QueueSizeGaugeMetric(sq.BackendName, sq.ClusterType, sq.Workqueue.Len())
+		sq.Metrics.QueueSizeGaugeMetric(sq.Workqueue.Len())
 		sq.Logger.Errorf("got an unknown item of type %T in the queue", obj)
 		return true
 	}
 
-	err := action.Sync(sq.KubeClient, sq.Metrics, sq.BackendName)
+	err := action.Sync(sq.KubeClient, sq.Logger)
+	if err != nil {
+		sq.Metrics.ServiceMetricError(action.ObjectMeta().GetNamespace(), action.ObjectMeta().GetName(), action.GetActionType())
```

Comment:
What if we get an error while handling an Endpoints resource?

Reply:
yup, meant to refactor that, good catch.